### PR TITLE
directly parse peer IDs as peer IDs

### PR DESCRIPTION
**Pull request by Stebalien**
No need to parse it as a hash first.

### CODE REVIEW

**Comment by Stebalien** [comment]
@kevina this is one of the few to only changes we'll need to make to go-ipfs to use non-byte-array CIDs.

File: namesys/routing.go [hunk 1]
```diff
@@ -55,19 +55,13 @@ func (r *IpnsResolver) resolveOnce(ctx context.Context, name string, options *op
 	}
 
 	name = strings.TrimPrefix(name, "/ipns/")
-	hash, err := mh.FromB58String(name)
+	pid, err := peer.IDB58Decode(name)
```
Comment (kevina) [inline namesys/routing.go]: Unless I am missing something, this is not performing the same set of checks.  In particular peer.IDB58Decode does not check it is a valid Multihash: https://github.com/libp2p/go-libp2p-peer/blob/master/peer.go#L121:L127

Comment (Stebalien) [reply namesys/routing.go]: It calls `mh.FromB58String` internally.

Comment (kevina) [reply namesys/routing.go]: Oops sorry.

**Comment by Stebalien** [comment]
@kevina could I get a :heavy_check_mark:?

File: namesys/routing.go [hunk 2]
```diff
@@ -55,19 +55,13 @@ func (r *IpnsResolver) resolveOnce(ctx context.Context, name string, options *op
...
 	if err != nil {
 		// name should be a multihash. if it isn't, error out here.
 		log.Debugf("RoutingResolver: bad input hash: [%s]\n", name)
```
[the remark this answers is not in the document]
Comment (Stebalien) [reply namesys/routing.go]: Fixed (although I chose a different error message as that one isn't quite right either).

**Pull request merged**